debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/antlr/listener/ColumnDefinitionParserListener.java

```diff
@@ -193,7 +194,19 @@ else if (dataTypeContext instanceof MySqlParser.DimensionDataTypeContext) {
 
             if (dimensionDataTypeContext.lengthTwoOptionalDimension() != null) {
                 List<MySqlParser.DecimalLiteralContext> decimalLiterals = dimensionDataTypeContext.lengthTwoOptionalDimension().decimalLiteral();
-                length = Integer.valueOf(decimalLiterals.get(0).getText());
+                if (decimalLiterals.get(0).REAL_LITERAL() != null) {
+                    String[] digits = decimalLiterals.get(0).getText().split("\\.");
```

Comment:
Could you extract a `Pattern` and store it as a static field, avoiding the repeated parsing of the regexp?

Reply:
@gunnarmorling
I have fixed these issue with new pr:
https://github.com/debezium/debezium/pull/2692